host/lib/usrp/dboard_iface.cpp

```diff
@@ -19,7 +19,7 @@ void dboard_iface::sleep(const std::chrono::nanoseconds& time)
     // FIXME: Create a delay in the FPGA on the device.
     auto cmd_time = get_command_time();
     if (cmd_time.get_real_secs() != 0.0) {
-        set_command_time(cmd_time + uhd::time_spec_t(time.count()));
+        set_command_time(cmd_time + uhd::time_spec_t(static_cast<double>(time.count())));
```

Comment:
This will take the number of nanoseconds (e.g., 166ns) and return it as a double (166.0) and then add it to `cmd_time` -- potentially as many seconds in the future as nanoseconds where requested. The same is true for the original code.